canonical/vanilla-framework
# Full width dropdowns for Vanilla "meganav" navigation

## Done

This PR adds a new style of top navigation dropdowns that are full-width containers.
Mobile version still uses the same "sliding" navigation lists of items, but desktop dropdowns will contain a separate content, with full support for grid and Vanilla components or any custom HTML.

NOTE: this feature is part of bigger work in progress for upstreaming meganav styling and is only meant to be used in the context of meganav on canonical.com or other websites.



Fixes https://warthogs.atlassian.net/browse/WD-12784

## QA

**NOTE**: the functionality is only about dropdown container, the **content of the dropdown is not part of this feature**. For demo we have some test content that uses grid.

- Open [demo](https://vanilla-framework-5213.demos.haus/docs/examples/patterns/navigation/dropdown-full-width?theme=light)
- Full width dropdowns on "home page" top navigation
https://vanilla-framework-5213.demos.haus/docs/examples/patterns/navigation/dropdown-full-width?theme=light
- Full width dropdowns on reduced navigation:
https://vanilla-framework-5213.demos.haus/docs/examples/patterns/navigation/reduced?theme=light
- Test both examples above 
  - make sure they open and close as expected
  - make sure it interacts well with search
  - make sure switching between dropdowns works
  - make sure on small screens sliding nav works as before (having it's own separate content)
- Review updated documentation:
  - https://vanilla-framework-5213.demos.haus/docs/patterns/navigation#full-width-dropdowns
  - (make sure you view on screen large enough to have embedded example show the dropdowns)

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)


## Screenshots

<img width="1771" alt="image" src="https://github.com/user-attachments/assets/3c973e6d-ed90-4990-991e-84e40d6c93c7">
